webbeans-tomcat/src/main/java/org/apache/webbeans/web/tomcat/TomcatPlugin.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.webbeans.web.tomcat;
+
+import javax.servlet.Filter;
+import javax.servlet.Servlet;
+import javax.servlet.ServletContextAttributeListener;
+import javax.servlet.ServletContextListener;
+import javax.servlet.ServletRequestAttributeListener;
+import javax.servlet.ServletRequestListener;
+import javax.servlet.http.HttpSessionActivationListener;
+import javax.servlet.http.HttpSessionAttributeListener;
+import javax.servlet.http.HttpSessionBindingListener;
+import javax.servlet.http.HttpSessionListener;
+
+import java.util.EventListener;
+
+import org.apache.tomcat.util.res.StringManager;
+import org.apache.webbeans.config.WebBeansContext;
+import org.apache.webbeans.exception.WebBeansConfigurationException;
+import org.apache.webbeans.spi.SecurityService;
+import org.apache.webbeans.spi.plugins.AbstractOwbPlugin;
+
+/**
+ * Tomcat plugin for OpenWebBeans.
+ */
+public class TomcatPlugin extends AbstractOwbPlugin {
+
+    private static final StringManager sm = StringManager.getManager(TomcatPlugin.class);
+
+    /**
+     * Security service implementation.
+     */
+    private final TomcatSecurityService securityService = new TomcatSecurityService(WebBeansContext.getInstance());
```

Comment:
looks fishy, why not adding a constructor with the webbeanscontext and reading the security service from this last one? Here you will get 2 instances (not a big deal yet with current impl but can be after some updates).